code/modules/research/techweb/_techweb.dm

```diff
@@ -415,7 +415,21 @@
 /datum/techweb/specialized/autounlocking/biogenerator
 	design_autounlock_buildtypes = BIOGENERATOR
 	allowed_buildtypes = BIOGENERATOR
-
+	var/bio_gen_part_tier = 0
+
+/datum/techweb/specialized/autounlocking/biogenerator/New()
+	spawn(10) 						//Waiting 1 second to get sent the part tier.
+		switch(bio_gen_part_tier) 	//This delay does cause a one second "blank screen" on the biogenerator, but goes away.
+			if(1)
+				design_autounlock_categories = list("initial")
+			if(2)
+				design_autounlock_categories = list("initial", "tier_two")
+			if(3)
+				design_autounlock_categories = list("initial", "tier_two", "tier_three")
+			if(4)
+				design_autounlock_categories = list("initial", "tier_two", "tier_three", "tier_four")
+		..()
+		autounlock()
```

Comment:
Do you really need spawn here? it seems unnecessary. New always gets called when its first made, and I don't see anything to set bio_gen_part_tier to anything but zero. It would be much better if the delay wasn't there in the first place, and its definitely doable.

Reply:
Without the delay, it runs too fast to actually update the tier.
The bio_gen_part_tier variable is changed by the biogenerator sending the tier of part along to this

Reply:
Maybe have this run after it sends the tier?